gnocchi/rest/api.py

```diff
@@ -1858,6 +1868,18 @@ def get_cross_metric_measures_from_objs(metrics, start=None, stop=None,
                          for metric in metrics),
                     'No granularity match'))
 
+        aggregations = set()
+        for metric in metrics:
+            for g in granularity:
```

Comment:
this duplicates what is happening on new line 1912. probably remove it from there?

Reply:
That only checks the granularity, not the method.

Reply:
```
                 if (aggregation
                    not in metric.archive_policy.aggregation_methods):
                     abort(404, {
                         "cause":
                         "Aggregation method does not exist for this metric",
                         "detail": {
                             "metric": str(metric.id),
                             "aggregation_method": aggregation,
                         },
                     })
```

that really seems to be checking method :P

Reply:
Not sure I understand. What I mean is that the piece of code I added only returns 404 if the granularity does not exist. But if the method does not exist it won't return 404.

The piece of code you pasted do this.

We need both.

Reply:
the code i pasted was what i was suggesting is being duplicated but i guess we can refactor in next.